aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp

```diff
@@ -676,8 +676,12 @@ static void qprelu_out_kernel(Tensor& out,
   DimVector sizes(input_ndim, 1), strides(input_ndim, 0);
   auto as_nd = [&](const Tensor& t) {
     TORCH_INTERNAL_ASSERT(t.defined() && (t.dim() == 1 || t.dim() == 0));
-    sizes[1] = t.dim() == 1 ? t.sizes()[0] : 1;
-    strides[1] = t.dim() == 1 ? t.strides()[0] : 0;
+    if (input_ndim > 1) {
+      // Take t's length and stride if it's 1d.
+      // Otherwise t is a scalar. Then size is 1 and stride is 0.
```

Comment:
can we explain why do we need to do this instead of what we did?

Reply:
looks like this is a broadcasting operation, can we just call broadcast_to: https://pytorch.org/docs/stable/generated/torch.broadcast_to.html?highlight=broadcast#torch.broadcast_to?
and if `t` is a 1d tensor, calling `t.numel()` should give you the number of elements I think

Reply:
Sorry but what do you mean by 'what we did'?

Reply:
looks like the comment is explaining what the code is doing, but didn't explain why we are doing this

Reply:
There are comments above these lines https://github.com/pytorch/pytorch/blob/e81aff7ff92e8e53a946f76184cb95f1b1352a3f/aten/src/ATen/native/quantized/cpu/kernels/QuantizedOpKernels.cpp#L674. Are these what you are looking for?

Reply:
yeah, I understand this better after looking at this comment, so suggested to do https://github.com/pytorch/pytorch/pull/82393/files#r936239329 here

Reply:
Looks like `boradcast_to` does not do the same thing as this lambda expression. For example, we have a 1-d tensor `weight = [1, 2, 3]` and we want to make it be of shape `(1, 3, 1, 1)`, i.e., `weight_4d = [[[[1]]], [[[2]]], [[[3]]]]`. But `broadcast` to cannot do that.
```shell
>>> import torch
>>> x = torch.tensor([1, 2, 3])
>>> torch.broadcast_to(x, (1, 3, 1, 1))
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
RuntimeError: The expanded size of the tensor (1) must match the existing size (3) at non-singleton dimension 3.  Target sizes: [1, 3, 1, 1].  Tensor sizes: [3]
```

Reply:
I see, does reshape work?

Reply:
Good idea. I tried and it looked good. Please check.